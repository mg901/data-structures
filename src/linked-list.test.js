const { LinkedListNode, LinkedList } = require('./linked-list');

describe('LinkedListNode', () => {
  const node = new LinkedListNode(1);
  test('returns correct value', () => {
    expect(node.value).toEqual(1);
  });

  test('returns default next node', () => {
    expect(node.next).toBeNull();
  });

  test('returns correct value from toString method', () => {
    expect(node.toString()).toEqual('1');
  });
});

describe('LinkedList', () => {
  test('returns the default values for the head and tail props', () => {
    const list = new LinkedList();

    expect(list.head).toBeNull();
    expect(list.tail).toBeNull();
  });

  describe('append method', () => {
    test('returns correct head and tail if append one node', () => {
      const list = new LinkedList();
      const node = {
        value: 1,
        next: null,
      };

      expect(list.append(1)).toEqual({
        head: node,
        tail: node,
      });
    });

    test('returns list with multiple nodes', () => {
      const list = new LinkedList();

      list.append(1).append(2);
      expect(list).toMatchSnapshot();
    });
  });

  test('toArray method', () => {
    const list = new LinkedList();

    list.append(1).append(2);
    expect(list.toArray()).toMatchSnapshot();
  });

  test('toString method', () => {
    const list = new LinkedList();

    list.append(1).append(2);

    expect(list.toString()).toEqual('1,2');
  });
});
