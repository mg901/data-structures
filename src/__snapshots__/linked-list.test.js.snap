// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedList  toArray method 1`] = `
[
  LinkedListNode {
    "next": LinkedListNode {
      "next": null,
      "value": 2,
    },
    "value": 1,
  },
  LinkedListNode {
    "next": null,
    "value": 2,
  },
]
`;

exports[`LinkedList append method foo 1`] = `
[
  LinkedListNode {
    "next": LinkedListNode {
      "next": null,
      "value": 2,
    },
    "value": 1,
  },
  LinkedListNode {
    "next": null,
    "value": 2,
  },
]
`;

exports[`LinkedList append method returns list with multiple nodes 1`] = `
LinkedList {
  "head": LinkedListNode {
    "next": LinkedListNode {
      "next": null,
      "value": 2,
    },
    "value": 1,
  },
  "tail": LinkedListNode {
    "next": null,
    "value": 2,
  },
}
`;

exports[`LinkedList append method toArray method 1`] = `
[
  LinkedListNode {
    "next": LinkedListNode {
      "next": null,
      "value": 2,
    },
    "value": 1,
  },
  LinkedListNode {
    "next": null,
    "value": 2,
  },
]
`;

exports[`LinkedList toArray method 1`] = `
[
  LinkedListNode {
    "next": LinkedListNode {
      "next": null,
      "value": 2,
    },
    "value": 1,
  },
  LinkedListNode {
    "next": null,
    "value": 2,
  },
]
`;
